olp-cpp-sdk-core/include/olp/core/client/OlpClientSettings.h

```diff
@@ -122,6 +127,18 @@ struct CORE_API AuthenticationSettings {
    */
   TokenProviderCallback provider;
 
+  /**
+   * @brief The user-provided function that returns 'ApiKey'.
+   *
+   * If this provider is set, it is used instead of the token provider.
+   * The returned value, if not empty, is added as a URL parameter
+   * to each request.
+   *
+   * @note This method must be synchronized and should not trigger any tasks on
+   * 'TaskScheduler' as this might result in a deadlock.
```

Comment:
```suggestion
   * `TaskScheduler` as this might result in a deadlock.
```
Please use the acute accent (`) not the apostrophe (').